sdk/core/System.ClientModel/src/Convenience/ClientCache.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.ClientModel.Primitives;
+
+/// <summary>
+/// A cache for storing client instances, ensuring efficient reuse.
+/// Implements an LRU eviction policy.
+/// </summary>
+public class ClientCache
+{
+    private readonly ConcurrentDictionary<(Type, string), (object Client, long LastUsed)> _clients = new();
+
+    private const int MaxCacheSize = 100;
+
+    /// <summary>
+    /// Retrieves a client from the cache or creates a new one if it doesn't exist.
+    /// Updates the last-used timestamp on every hit.
+    /// </summary>
+    /// <typeparam name="T">The type of the client.</typeparam>
+    /// <param name="createClient">A factory function to create the client if not cached.</param>
+    /// <param name="id">An optional identifier for the client instance.</param>
+    /// <returns>The cached or newly created client instance.</returns>
+    public T GetClient<T>(Func<T> createClient, string id = "") where T : class
+    {
+        (Type, string) key = (typeof(T), id ?? string.Empty);
+        lock (_clients)
```

Comment:
should we use RW lock here as well? This could allow multiple readers at a time.